rasterio/__init__.py

```diff
@@ -236,7 +238,13 @@ def open(fp, mode='r', driver=None, width=None, height=None, count=None,
     # dataset object that we will return. When a dataset's close method
     # is called, this ExitStack will be unwound and the MemoryFile's
     # storage will be cleaned up.
-    if mode == 'r' and hasattr(fp, 'read'):
+    if opener:
```

Comment:
Should this specifically check `opener not None`? or perhapse even `callable(opener)`?

Reply:
@groutr I looked at Python's source and it doesn't go this far. I think it's okay, for now at least, to just fail if opener isn't callable.